vortex-buffer/src/lib.rs

```diff
@@ -42,31 +41,10 @@ impl Buffer {
         }
     }
 
-    pub fn typed_data<T: NativePType>(&self) -> &[T] {
+    pub fn into_vec(self) -> Result<Vec<u8>, Buffer> {
         match self {
-            Buffer::Arrow(buffer) => unsafe {
-                match_each_native_ptype!(T::PTYPE, |$T| {
-                    std::mem::transmute(buffer.typed_data::<$T>())
-                })
-            },
-            Buffer::Bytes(bytes) => {
-                // From ArrowBuffer::typed_data
-                let (prefix, offsets, suffix) = unsafe { bytes.align_to::<T>() };
-                assert!(prefix.is_empty() && suffix.is_empty());
-                offsets
-            }
-        }
-    }
-
-    pub fn into_vec<T: NativePType>(self) -> Result<Vec<T>, Buffer> {
-        match self {
-            Buffer::Arrow(buffer) => match_each_native_ptype!(T::PTYPE, |$T| {
-                buffer
-                    .into_vec()
-                    .map(|vec| unsafe { std::mem::transmute::<Vec<$T>, Vec<T>>(vec) })
-                    .map_err(Buffer::Arrow)
-            }),
-            // Cannot always convert bytes into a mutable vec
+            Buffer::Arrow(buffer) => buffer.into_vec::<u8>().map_err(Buffer::Arrow),
+            // Cannot convert bytes into a mutable vec
             Buffer::Bytes(_) => Err(self),
```

Comment:
why is this an Err? If you are the only one holding the bytes then this should be possible

Reply:
I don't think you can go from Bytes back to BytesMut

Reply:
https://rustasync.github.io/runtime/bytes/struct.Bytes.html#method.try_mut seems to be the right signature and then you can use that to construct vec from parts?

Reply:
🤦 I missed that. I thought it seemed odd

Reply:
Hmmm no, I'm not sure what docs those are but they're not up-to-date for the bytes crate. See also https://github.com/tokio-rs/bytes/issues/350

Reply:
Looks like someone is hosting their own fork or some outdated link that google served. You’re right. Maybe we need our own version as well?

Reply:
We do anyway to have more control over alignment. This crate tries to abstract that away for now.